### PR TITLE
numpy: downgrade setuptools

### PR DESCRIPTION
**Pull request by gromgit**
Upstream confirms that `numpy` requires setuptools < 60.0.

Also fully-qualifying `gfortran` path, because my local build env couldn't find it for some reason.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

File: Formula/numpy.rb [hunk 1]
```diff
@@ -38,9 +46,15 @@ def install
 
     xy = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
     ENV.prepend_create_path "PYTHONPATH", Formula["cython"].opt_libexec/"lib/python#{xy}/site-packages"
+    ENV.prepend_path "PYTHONPATH", libexec/"vendor/lib/python#{xy}/site-packages"
```
Comment (carlocab) [inline Formula/numpy.rb:49]:
```suggestion
    ENV.prepend_path "PYTHONPATH", libexec/"vendor"/Language::Python.site_packages("python3")
```
Is this not only needed at build-time?

Comment (gromgit) [reply Formula/numpy.rb:49]: It should only be necessary at build-time, but being a complete Python n00b, I cargo-culted this off the `yle-dl` formula. Is there a better solution?

**Comment by gromgit** [comment]
https://github.com/Homebrew/homebrew-core/runs/4923377804?check_suite_focus=true#step:8:406

This `bzt` test error is also happening locally with the original `numpy` bottle. I've tried rebuilding `bzt` on M1 Monterey, but that failed miserably too: https://gist.github.com/gromgit/de5e26d40ac2b173964834fc3962a22e

File: Formula/numpy.rb (hunk 1 above)
Comment (gromgit) [reply Formula/numpy.rb]: OK, using `buildpath/"temp"` instead of `libexec/"vendor"` seems to work.

**Comment by carlocab** [comment]
Yea, `bzt` has been spotted elsewhere.

File: Formula/numpy.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@ class Numpy < Formula
   url "https://files.pythonhosted.org/packages/0a/c8/a62767a6b374a0dfb02d2a0456e5f56a372cdd1689dbc6ffb6bf1ddedbc0/numpy-1.22.1.zip"
   sha256 "e348ccf5bc5235fc405ab19d53bec215bb373300e5523c7b476cc0da8a5e9973"
   license "BSD-3-Clause"
+  revision 1
```
Comment (carlocab) [inline Formula/numpy.rb:7]: A `revision` is not necessary, no? Since this was probably last built with the right version of `setuptools`.

Comment (gromgit) [reply Formula/numpy.rb]: Oh, right. :hand_over_mouth:

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request merged**